### PR TITLE
Add option to hide the "playback" message

### PR DESCRIPTION
**Pull request by Valinwolf**
PR that implements #12

### CODE REVIEW

**Comment by kgraefe** [comment]
Thanks for your contribution. I think that should be a plugin preference rather than a account preference, though (see e.g. the [offlinemsg plugin][1]).
 
[1]: https://bitbucket.org/pidgin/main/src/d4d7095686cf871867b0ff602b5cd7ae7737c8df/libpurple/plugins/offlinemsg.c#lines-240

File: src/znchelper.c [hunk 1]
```diff
@@ -98,6 +98,7 @@ static void znc_write_chat(
 	PurpleConversation *conv, const char *who, const char *message,
 	PurpleMessageFlags flags, time_t mtime
 ) {
+	PurpleConnection *gc = purple_conversation_get_gc(conv);
```
Comment (kgraefe) [inline src/znchelper.c]: Also please don't mix up declarations and code and make sure to check the return values.

Comment (Valinwolf) [reply src/znchelper.c]: I'll fix that right now.

**Comment by Valinwolf** [comment]
> Thanks for your contribution. I think that should be a plugin preference rather than a account preference, though (see e.g. the [offlinemsg plugin](https://bitbucket.org/pidgin/main/src/d4d7095686cf871867b0ff602b5cd7ae7737c8df/libpurple/plugins/offlinemsg.c#lines-240)).

I actually agree with you, but I was only using the code I had (this plugin) to formulate the code. I'm not very well versed in C. I can do Java, Python, PHP, and more all day, but C and friends are not a strong suit for me. I'll look into fixing that.

File: src/znchelper.c (hunk 1 above)
Comment (Valinwolf) [reply src/znchelper.c]: commit added to fix this

**Comment by Valinwolf** [comment]
All discussed changes have been made. Commits are ready for re-review.

File: src/znchelper.c [hunk 2]
```diff
@@ -37,6 +37,8 @@ static gboolean core_quitting = FALSE;
 #define ZNC_CONV_STATE_START 0
 #define ZNC_CONV_STATE_REPLAY 1
 #define ZNC_CONV_STATE_DONE 2
+#define PREF_PREFIX "/plugins/core/znc-helper"
+#define	PREF_HIDEMSG PREF_PREFIX "/hidemsg"
```
Comment (kgraefe) [inline src/znchelper.c]: This Tab should be a space.

File: src/znchelper.c [hunk 3]
```diff
@@ -102,7 +104,7 @@ static void znc_write_chat(
 	struct znc_conn *znc;
 	GList *parted, *l;
 	gint state;
-
+
```
Comment (kgraefe) [inline src/znchelper.c]: Remove whitespace

File: src/znchelper.c [hunk 4]
```diff
@@ -463,6 +469,7 @@ static gboolean plugin_load(PurplePlugin *plugin) {
 		_("Uses ZNC bouncer"), "uses_znc_bouncer", FALSE
 	);
 	irc_info->protocol_options = g_list_append(irc_info->protocol_options, option);
+
```
Comment (kgraefe) [inline src/znchelper.c]: Remove newline

File: src/znchelper.c [hunk 5]
```diff
@@ -540,6 +547,33 @@ static gboolean plugin_unload(PurplePlugin *plugin) {
...
 }
 
+static PurplePluginPrefFrame *
+get_plugin_pref_frame(PurplePlugin *plugin)
+{
+	PurplePluginPrefFrame *frame;
+	PurplePluginPref *pref;
+
+	frame = purple_plugin_pref_frame_new();
+
```
Comment (kgraefe) [inline src/znchelper.c]: Remove tab

**Comment by kgraefe** [comment]
LGTM apart from a few whitespace issues.

**Comment by kgraefe** [comment]
Thank you.

**Pull request merged**